sdk/src/main/java/com/hedera/hashgraph/sdk/PrivateKey.java

```diff
@@ -78,7 +81,14 @@ public static PrivateKey generateECDSA() {
      * @return      the ED25519 private key
      */
     public static PrivateKey fromSeedED25519(byte[] seed) {
-        return PrivateKeyED25519.fromSeed(seed);
+        var hmacSha512 = new HMac(new SHA512Digest());
```

Comment:
Putting code particular to a key algorithm into the base class doesn't seem right to me. I'll leave path forward up to you since I'm not involved in the architecture, but IMO this needs a refactor